plugins/BEdita/Core/src/Model/Table/UsersTable.php

```diff
@@ -295,6 +302,73 @@ public function beforeDelete(Event $event, EntityInterface $entity)
         }
     }
 
+    /**
+     * Modify user entity to become anonymous and hidden
+     *
+     * @param \Cake\Datasource\EntityInterface $entity the entity to anonimize
+     * @return void
+     */
+    protected function anonymizeUser(EntityInterface $entity)
+    {
+        $notNull = $this->notNullableColumns($this);
+        foreach ($this->inheritedTables() as $table) {
+            $notNull = array_merge($notNull, $this->notNullableColumns($table));
+        }
+        $properties = array_diff((array)$entity->visibleProperties(), $notNull, ['type']);
+        foreach ($properties as $name) {
+            $entity->set($name, null);
+        }
+        $deletedValue = self::DELETED_USER_PREFIX . $entity->get('id');
+        $entity->set('username', $deletedValue);
+        $entity->set('uname', $deletedValue);
+        $entity->set('password', null);
+        $entity->set('deleted', true);
+        $entity->set('locked', true);
+        $this->saveOrFail($entity);
+    }
+
+    /**
+     * Retrieve not nullable columns for a table
+     *
+     * @param \Cake\ORM\Table $table Table class
+     * @return array
+     */
+    protected function notNullableColumns($table)
+    {
+        $res = [];
+        $schema = $table->getSchema();
+        $columns = $schema->columns();
+        foreach ($columns as $col) {
+            if (!$schema->isNullable($col)) {
+                $res[] = $col;
+            }
+        }
+
+        return $res;
+    }
+
+    /**
+     * Override Table delete: in case of constraints avoid delete and anonymize user data
+     *
+     * @param \Cake\Datasource\EntityInterface $entity the entity to delete
+     * @param array $options Additional options.
+     * @return bool
+     */
+    public function delete(EntityInterface $entity, $options = [])
+    {
+        $exists = $this->exists([
+            'OR' => ['created_by' => $entity->get('id'), 'modified_by' => $entity->get('id')],
+        ]);
+        if (!$exists) {
+            return parent::delete($entity, $options);
+        }
+
+        $this->beforeDelete(new Event('Model.beforeDelete'), $entity);
+        $this->anonymizeUser($entity);
```

Comment:
(continues from above) …because here we could `return $this->anonymizeUser($entity);`.

This would make `delete()` behaviour more consistent to the original one. One may still use `$this->deleteOrFail()` in case they want an exception to be raised upon failure.